### PR TITLE
T migration modals (Undelegate + Claim)

### PR DESCRIPTION
**Pull request by r-czajkowski**
Depends on: #2670 

This PR adds new modals for the `Undelegate`  and `Claim` flow according to the Figma views.

<details>
  <summary>Screenshots</summary>
  
  obraz
  
  
obraz

obraz

obraz

obraz

obraz

</details>

### CODE REVIEW

File: solidity-v1/dashboard/src/components/Undelegations.jsx [hunk 1]
```diff
@@ -91,6 +91,7 @@ const Undelegations = ({ undelegations, title }) => {
                     undelegation.managedGrantContractInstance
                   }
                   operatorAddress={undelegation.operatorAddress}
+                  amount={undelegation.amount}
```
Comment (michalsmiarowski) [inline solidity-v1/dashboard/src/components/Undelegations.jsx:94]: Can we also change button text of the recover button here? I think we decided to change it to 'Claim' (and it is like this also in Figma)

`btnText={"claim"}`

File: solidity-v1/dashboard/src/components/modal/staking/claim/TokensClaimed.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+          <span className="ml-1">Success!</span>
+        </h3>
+        <h4 className="text-grey-70 mb-2">
+          View your transaction&nbsp;
+          <ViewInBlockExplorer
+            type="tx"
+            className="text-grey-70"
+            id={txHash}
+            text="here"
+          />
```
Comment (michalsmiarowski) [inline solidity-v1/dashboard/src/components/modal/staking/claim/TokensClaimed.jsx:25]: There is no arrow in FIgma next to `here` link. Can we add the posisbility to remove it in `ViewInBlockExplorer` component?

File: solidity-v1/dashboard/src/components/modal/staking/undelegate/UndelegationInitiated.jsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+          <span className="ml-1">Success!</span>
+        </h3>
+        <h4 className="text-grey-70 mb-2">
+          View your transaction&nbsp;
+          <ViewInBlockExplorer
+            type="tx"
+            className="text-grey-70"
+            id={txHash}
+            text="here"
+          />
```
Comment (michalsmiarowski) [inline solidity-v1/dashboard/src/components/modal/staking/undelegate/UndelegationInitiated.jsx:31]: Same as above

**Review by michalsmiarowski — CHANGES_REQUESTED** [review]
Overall looking very good! Left some comments to look at

File: solidity-v1/dashboard/src/components/RecoverStakeButton.jsx [hunk 4]
```diff
@@ -1,39 +1,42 @@
...
+  isFromGrant,
+  ...props
+}) => {
+  const { openConfirmationModal, openModal } = useModal()
+  const address = useWeb3Address()
+
+  const onRecoverStake = async () => {
+    const { tokenStakingEscrow } = await ContractsLoaded
+    let destinationAddress = address
+    console.log("RecoverStakeButton")
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply solidity-v1/dashboard/src/components/RecoverStakeButton.jsx]: 84536fa

File: solidity-v1/dashboard/src/reducers/staking.js [hunk 5]
```diff
@@ -136,6 +136,7 @@ const removeFromDelegationOrUndelegation = (array, id) => {
     array,
     compareEthAddresses
   )
+  console.log("indexInArray", array, id)
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply solidity-v1/dashboard/src/reducers/staking.js]: 84536fa

File: solidity-v1/dashboard/src/components/Undelegations.jsx (hunk 1 above)
Comment (r-czajkowski) [reply solidity-v1/dashboard/src/components/Undelegations.jsx:94]: b2bf118

File: solidity-v1/dashboard/src/components/modal/staking/claim/TokensClaimed.jsx (hunk 2 above)
Comment (r-czajkowski) [reply solidity-v1/dashboard/src/components/modal/staking/claim/TokensClaimed.jsx:25]: I left a comment about this in [Figma](https://www.figma.com/file/Su61MwHLrQn3ld24q85pMx?node-id=11784:87075#124245039) cc @liz-shinn @sashatanase. This is an external link and for any other links we always display an arrow indicating that it is an external link.

Comment (liz-shinn) [reply solidity-v1/dashboard/src/components/modal/staking/claim/TokensClaimed.jsx:25]: Yup! Will make that update in the Figma file. 

There were a few other things I tweaked as well - all of the modal button text and titles were default ones (I believe they were overridden at some point by changes to the master component in the DS file) so I corrected those to what they should have been.

**Review by michalsmiarowski — APPROVED** [review]
Looking good!

**Pull request merged**